src/ray/common/client_connection.cc

```diff
@@ -199,10 +200,12 @@ template <class T>
 ClientConnection<T>::ClientConnection(MessageHandler<T> &message_handler,
                                       boost::asio::basic_stream_socket<T> &&socket,
                                       const std::string &debug_label,
+                                      const char *const *message_type_enum_names,
```

Comment:
Let's use a std::vector<std::string> here to make the memory management more explicit (if it's only constructed/copied on initialization of ClientConnection, the performance impact should be minimal)

Reply:
Sure, I actually only did this because this is already provided in the generated flatbuffer files, but I can copy it over to a `std::vector`.

Reply:
Ah ok, if it is statically allocated and already there, I'd be ok with it, but then let's also pass in the length of the whole vector and document it to make sure the reader knows what's going on. The vector solution is also ok.